internal/services/managedapplications/managed_application_definition_resource.go

```diff
@@ -133,55 +134,51 @@ func resourceManagedApplicationDefinitionCreateUpdate(d *pluginsdk.ResourceData,
+	id := applicationdefinitions.NewApplicationDefinitionID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
 
 	if d.IsNewResource() {
-		existing, err := client.Get(ctx, id.ResourceGroup, id.Name)
+		existing, err := client.Get(ctx, id)
 		if err != nil {
-			if !utils.ResponseWasNotFound(existing.Response) {
+			if !response.WasNotFound(existing.HttpResponse) {
 				return fmt.Errorf("failed to check for presence of existing %s: %+v", id, err)
 			}
 		}
-		if !utils.ResponseWasNotFound(existing.Response) {
+		if !response.WasNotFound(existing.HttpResponse) {
 			return tf.ImportAsExistsError("azurerm_managed_application_definition", id.ID())
 		}
 	}
 
-	parameters := managedapplications.ApplicationDefinition{
-		Location: utils.String(azure.NormalizeLocation(d.Get("location"))),
-		ApplicationDefinitionProperties: &managedapplications.ApplicationDefinitionProperties{
+	parameters := applicationdefinitions.ApplicationDefinition{
+		Location: pointer.To(azure.NormalizeLocation(d.Get("location"))),
+		Properties: applicationdefinitions.ApplicationDefinitionProperties{
 			Authorizations: expandManagedApplicationDefinitionAuthorization(d.Get("authorization").(*pluginsdk.Set).List()),
-			Description:    utils.String(d.Get("description").(string)),
-			DisplayName:    utils.String(d.Get("display_name").(string)),
+			Description:    pointer.To(d.Get("description").(string)),
+			DisplayName:    pointer.To(d.Get("display_name").(string)),
 			IsEnabled:      utils.Bool(d.Get("package_enabled").(bool)),
-			LockLevel:      managedapplications.ApplicationLockLevel(d.Get("lock_level").(string)),
+			LockLevel:      applicationdefinitions.ApplicationLockLevel(d.Get("lock_level").(string)),
 		},
 		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
 	}
 
 	if v, ok := d.GetOk("create_ui_definition"); ok {
-		parameters.CreateUIDefinition = utils.String(v.(string))
+		parameters.Properties.CreateUiDefinition = &v
 	}
 
 	if v, ok := d.GetOk("main_template"); ok {
-		parameters.MainTemplate = utils.String(v.(string))
+		parameters.Properties.MainTemplate = &v
 	}
 
-	if (parameters.CreateUIDefinition != nil && parameters.MainTemplate == nil) || (parameters.CreateUIDefinition == nil && parameters.MainTemplate != nil) {
+	if (parameters.Properties.CreateUiDefinition != nil && parameters.Properties.MainTemplate == nil) || (parameters.Properties.CreateUiDefinition == nil && parameters.Properties.MainTemplate != nil) {
 		return fmt.Errorf("if either `create_ui_definition` or `main_template` is set the other one must be too")
 	}
 
 	if v, ok := d.GetOk("package_file_uri"); ok {
-		parameters.PackageFileURI = utils.String(v.(string))
+		parameters.Properties.PackageFileUri = pointer.To(v.(string))
 	}
 
-	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, parameters)
-	if err != nil {
+	if _, err := client.CreateOrUpdate(ctx, id, parameters); err != nil {
```

Comment:
this was a Future, so should this be:

```suggestion
	if err := client.CreateOrUpdateThenPoll(ctx, id, parameters); err != nil {
```

Reply:
There is no polling function in the new SDK